packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -584,6 +592,14 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
   }
 
   void _handleSingleTapUp(TapUpDetails details) {
+    // Because TextSelectionGestureDetector listens to taps that happen on
+    // widgets in front of it, tapping the clear button will also trigger this
+    // handler here. If this is the case, return.
+    if (_clearing) {
```

Comment:
This is hacky.  Is there any other way to prevent taps on the clear button from triggering the TextSelectionGestureDetector?  I'm back to the confusing gesture arena problem of multiple widgets getting the same event.

Reply:
Ideally we want to eliminate gestures that start (with a down event) in either the text field's prefix or the clear button suffix. The TextSelectionGestureDetector would have to ignore gestures that start within the corresponding render boxes, per [hitTest](https://api.flutter.dev/flutter/rendering/RenderBox/hitTest.html).

This probably isn't academic. One could use a Slider as prefix for example.